docs_src/microservices/application/ApplicationServiceAPI.md

```diff
@@ -260,22 +271,60 @@ See the [App Service Template](https://github.com/edgexfoundry/app-functions-sdk
 
 The following `ApplicationService` APIs allow your service to set the Functions Pipeline and start and stop the Functions Pipeline.
 
+### AppFunction
+
+`type AppFunction = func(appCxt AppFunctionContext, data interface{}) (bool, interface{})`
+
+This type defines the signature that all pipeline functions must implement.
+
+### FunctionPipeline
+
+This type defines the struct that contains the metadata for a functions pipeline instance.
+
+```go
+type FunctionPipeline struct {
+	Id         string
+	Transforms []AppFunction
+	Topic      string
+	Hash       string
+}
+```
+
 ### SetFunctionsPipeline
 
 `SetFunctionsPipeline(transforms ...AppFunction) error`
 
-This API sets up the functions pipeline with the specified list of Application Functions. Note that the functions are executed in the order provided in the list.  An error is returned if the list is empty.
+This API sets up the default functions pipeline with the specified list of Application Functions. Note that the functions are executed in the order provided in the list.  An error is returned if the list is empty. Useful when only one functions pipeline is needed
 
 !!! example "Example - SetFunctionsPipeline"
     ```go
     sample := functions.NewSample()
-    err = service.SetFunctionsPipeline(
-        transforms.NewFilterFor(deviceNames).FilterByDeviceName,
-        sample.LogEventDetails,
-        sample.ConvertEventToXML,
-        sample.OutputXML)
+    err = service.SetFunctionsPipeline(transforms.NewFilterFor(deviceNames).FilterByDeviceName,
+                                       sample.LogEventDetails,
+                                       sample.ConvertEventToXML,
+                                       sample.OutputXML)
+    if err != nil {
+        ...
+        return -1
+    }
+    ```
+
+### AddFunctionsPipelineForTopic
+
+`AddFunctionsPipelineForTopic(id string, topic string, transforms ...AppFunction) error`
+
+This API adds a functions pipeline with the specified unique ID and list of functions (transforms) to be executed when the received topic matches the pipeline topic. See the [Pipeline Per Topic](../AdvancedTopics/#pipeline-per-topic) section for more details.
+
+!!! example "Example - AddFunctionsPipelineForTopic"
+    ```go
+    sample := functions.NewSample()
+    err = service.AddFunctionsPipelineForTopic("Floats-Pipeline", "edgex/events/#/#/Random-Float-Device/#",
```

Comment:
I feel like we should standardize names either AddFunctionsPipeline* or SetFunctionsPipeline* not mix like this.